ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+/// Assign a map viw delegate, may be `nil`, see `TGMapViewDelegate` for more details
+@property (weak, nonatomic, nullable) id<TGMapViewDelegate> mapViewDelegate;
+
+#pragma mark Camera/View modifiers
+
+/* The following are computed property. They return sensible defaults when the above .map property is nil */
+
+/// Assign a `TGCameraType` to the view camera
+@property (assign, nonatomic) TGCameraType cameraType;
+/// Assign a longitude and latitude to the map view camera
+@property (assign, nonatomic) TGGeoPoint position;
+/// Assign a floating point zoom to the map view camera
+@property (assign, nonatomic) float zoom;
+/// Assign a rotation angle in radians to the map view camera
+@property (assign, nonatomic) float rotation;
+/// Assign a tilt angle in radians to the map view camera
+@property (assign, nonatomic) float tilt;
+
+/**
+ Assign `TGHttpHandler` for network request management.
+ A http handler with a default configuration is already provided to the map view.
+
+ Must be non-`nil`.
+
+ @note Assigning a http handler is optional and should only be done if you want to change
+ any network related configuration (cache path and network cache size, both memory and disk,
+ or be notified when a network request completes).
+ More informations on the default provided configuration can be found on the description
+ of `TGHttpHandler`.
+ */
+@property (strong, nonatomic) TGHttpHandler* httpHandler;
+
+#pragma mark Debug toggles
+
+/**
+ Set a `TGDebugFlag` to the map view
+
+ @param debugFlag the debug flag to set
+ @param value whether the flag is on or off
+ */
+- (void)setDebugFlag:(TGDebugFlag)debugFlag value:(BOOL)on;
+
+/**
+ Get the status of a specific `TGDebugFLag`
+
+ @param debugFlag the debug flag to get the status of
+ @return whether the flag is currently on or off for the map view
+ */
+- (BOOL)getDebugFlag:(TGDebugFlag)debugFlag;
+
+/**
+ Changes the status of a debug flag.
+ If a debug flag is on, this will turn it off and vice-versa.
+
+ @param debugFlag the debug flag to toggle the status of
+ */
+- (void)toggleDebugFlag:(TGDebugFlag)debugFlag;
+
+#pragma mark Marker interface
+
+/**
+ Removes all the markers added to the map view.
+ */
+- (void)markerRemoveAll;
+
+/**
+ Creates a marker identifier which can be used to dynamically add points and polylines
+ to the map.
+
+ Example on how to add a point to the map for a geographic coordinate.
+
+ ```objc
+ TGMapMarkerId identifier = [view markerAdd];
+ [view markerSetStyling:identifier styling:@"{ style: 'points', color: 'white', size: [25px, 25px], order:500 }"];
+ [view markerSetPoint:identifier coordinates:coordinate];
+ ```
+
+ @return a marker identifier
+ @note The marker should be appropriately styled using `-markerSetStyling:styling:`
+ and a type must be set (point, polyline, polygon) through `markerSetPoint*`, `markerSetPolyline` or `markerSetPolygon`
+ for it to be visible.
+
+ @note A marker identifier of 0 is non-valid.
```

Comment:
Might want to be a little more explanatory and indicate that 0 is returned when adding a marker to the map failed.